storage/src/main/java/tech/pegasys/teku/storage/archive/DataArchive.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Consensys Software Inc., 2024
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.storage.archive;
+
+import java.io.IOException;
+import java.util.List;
+import tech.pegasys.teku.spec.datastructures.blobs.versions.deneb.BlobSidecar;
+
+/**
+ * Interface for a data archive which stores non-current BlobSidecars and could be extended later to
```

Comment:
`non-current`, maybe something more specific like `prunable` or `BlobSidecars outside the Data Availability window`